node/Utils/CheckRLimits.hs

```diff
@@ -0,0 +1,39 @@
+{-# language ScopedTypeVariables #-}
+
+module Utils.CheckRLimits(checkRLimits) where
+
+import Control.Monad
+import Data.Char
+import Data.List(stripPrefix)
+import System.Exit
+import System.IO
+import System.Posix.Process (getProcessID)
+import Text.Read
+
+checkRLimits :: IO ()
+checkRLimits = do
+    pid <- getProcessID
+    limits <- readFile $ "/proc/" <> show pid <> "/limits"
```

Comment:
This won't work on mac-os and windows. Maybe use `getrlimit` and `setrlimit` with FFI, I think, I think are both POSIX.

Reply:
Also, if this fails, the error message would probably be confusing for most users.